### PR TITLE
Feat: version faq docs

### LINKED ISSUE #935: Provide versioning for our FAQ
Provide versioning for our FAQ, given the FAQ data is specific per version (ie supported Kubernetes version)

https://keda.sh/docs/2.9/faq/

### PR DESCRIPTION
**Pull request by thisisobate**
This PR introduces versioning for FAQ.

### Versions affected
v2.14 and upwards

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #935

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -143,13 +157,34 @@ Here are a few examples:
...
+
+```console
+$ cd content/docs/<NewVersion>/reference/faq.md
+```
+
+4. update the versionData option
+
+```
+{{< faq20 versionData="NEW_FAQ_FILE_NAME" >}}
+```
```
Comment (tomkerkhove) [inline CONTRIBUTING.md:184]: Massive nit, but please start sentence with capital letter

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -291,7 +329,7 @@ parse[doc.title] = {
   availability: doc.availability,
   category: doc.category,
   type: doc.type,
-  FILTER_NAME: doc.FILTER_NAME
+  FILTER_NAME: doc.FILTER_NAME,
```
Comment (tomkerkhove) [inline CONTRIBUTING.md:332]: Intended?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -143,13 +157,34 @@ Here are a few examples:
...
+1. copy the directory for the most recent version. Here's an example:
 
 ```console
 $ cp -rf content/docs/<CurrentVersion> content/docs/<NewVersion>
 ```
 
+2. copy the file for the most recent faq data in the `data` directory. Here's an example:
+
+```console
+$ cp -rf data/faq<CurrentVersion> data/faq<NewVersion>
```
Comment (tomkerkhove) [inline CONTRIBUTING.md:171]: Is there no way that the faq data is part of `content/docs/<version>`?

**Review by davogen — COMMENTED** [review]
I like this, I am new to KEDA and this is something that was strange to me in the FAQ section (due to a policy we need to use an older version of KEDA)

File: CONTRIBUTING.md (hunk 2 above)
Comment (thisisobate) [reply CONTRIBUTING.md:332]: no...prettier doing its thing

File: CONTRIBUTING.md (hunk 3 above)
Comment (thisisobate) [reply CONTRIBUTING.md:171]: unfortunately no.

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Thank you!

**Pull request merged**